pkg/api/types.go

```diff
@@ -910,6 +910,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// JobSpec is the specification of a job.
+// As the internal representation of a job, it may have either a TemplateRef or a Template set.
+type JobSpec struct {
+	// Completions is the number of desired successful completions of this job.
+	Completions int `json:"completions"`
+
+	// Selector is a label query over pods that should match the completed count.
+	Selector map[string]string `json:"selector"`
```

Comment:
It was discussed here:
https://github.com/GoogleCloudPlatform/kubernetes/issues/1624#issuecomment-95407993

There's no absolutely preventing multiple execution, since containers could always fail just before exiting with 0. I think the initial version could make a pretty-good effort at tracking completions in status. I had suggested persisting them in an annotation as a kind of private checkpoint.